clang/lib/CodeGen/CGObjCMac.cpp

```diff
@@ -3971,7 +3986,9 @@ void CGObjCCommonMac::GenerateDirectMethodPrologue(
     ReceiverCanBeNull = isWeakLinkedClass(OID);
```

Comment:
@MadCoder I noticed your TODO you left here five years ago, and I wonder if you want me to do something about it. With this patch we can easily move the initializer into the thunk. But I am not sure how to test if `self` is initialized or not.